chatkit-core/src/main/kotlin/com/pusher/chatkit/ChatManager.kt

```diff
@@ -65,8 +67,18 @@ class ChatManager(
     fun blocking() = syncChatManager
 }
 
-fun <V> makeCallback(f: () -> V, c: (V) -> Unit) {
+fun <V> makeCallback(f: () -> Result<V, Error>, c: (Result<V, Error>) -> Unit) {
     Futures.schedule {
-        c.invoke(f())
+        c.invoke(try {
+            f()
+        } catch (e: Throwable) {
+            Errors.other(e).asFailure<V, Error>()
+        })
+    }
+}
+
+fun <V> makeSingleCallback(f: () -> V, c: (V) -> Unit) {
```

Comment:
Perhaps it's a bit unsafe to have this function as it is -- it can have unexpected outcomes when there's an exception. And as discussed in Slack, potentially it could always return a `Result<V, Error>`.

I don't know how much would that would be so it could be done in a separate PR.

Reply:
This one remains because not all wrapped calls return a `Result`, and wrapping those calls in one would break compatibility (though I agree it is the ideal)